plugins/modem-manager/fu-mm-device.c

```diff
@@ -1550,10 +1552,10 @@ fu_mm_device_write_firmware_firehose(FuDevice *device,
 	/* Firehose program needs to be loaded to the modem before firehose update process can
 	 * start. Generally, modems use Sahara protocol to load the firehose binary.
 	 *
-	 * In case of MHI PCI modems, the mhi_wwan driver reads the firehose binary from the
+	 * In case of MHI PCI modems, the mhi-pci-generic driver reads the firehose binary from the
 	 * firmware-loader and writes it to the modem.
 	 **/
-	if (g_strrstr(fu_device_get_physical_id(device), "pci") && self->port_qcdm != NULL) {
+	if (g_strrstr(driver, "mhi-pci-generic") && self->port_qcdm != NULL) {
```

Comment:
Does g_strv_contains (iirc) work here?

Reply:
oh nice, didn't know about that. updated it in the commit